x-pack/plugins/canvas/public/components/function_form/index.tsx

```diff
@@ -80,9 +82,11 @@ export const FunctionForm: React.FunctionComponent<FunctionFormProps> = (props)
 
   const setArgument = useCallback(
     (argName: string, valueIndex: number) => (value: string | Ast | null) => {
-      dispatch(setArgumentValue({ element, pageId, argName, value, valueIndex, path }));
+      dispatch(
+        setArgumentValue({ elementId: element?.id, pageId, argName, value, valueIndex, path })
```

Comment:
It is required to use the `element?.id` to always have the fresh element with the expression AST of the latest version. It will prevent updating of the old expression with multiple args components and prevent causing the race conditions.